sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnnecessaryUsings.cs

```diff
@@ -151,6 +151,36 @@ public void DoLinqQuery(List<string> myList)
     }
 }
 
+namespace LinqQuery2
+{
+    using global::System.Linq; // Compliant - statement is needed for query syntax
+    class LinqQueryUser
+    {
+        public void DoLinqQuery(List<string> myList)
+        {
+            var linkQuery = from item in myList select item.GetType();
+        }
+    }
+}
+
+namespace NoLinqQuery
+{
+    using System.Linq; // Noncompliant
+    class UnusedLinqImport { }
+}
+
+namespace System
+{
+    using Linq; // Compliant - statement is needed for query syntax
+    class LinqQueryUser
+    {
+        public void DoLinqQuery(List<string> myList)
+        {
+            var linkQuery = from item in myList select item.GetType();
+        }
+    }
+}
+
```

Comment:
Please add test like this to be bullet-proof in the future:

```
namespace Custom.System.Linq
{
    class Coincidence
    {
    }
}

namespace Custom
{
    using System.Linq;  //Noncompliant, it is from Custom.System.Linq and it is not used.
    using global::System.Linq;

    class Usage
    {
        public void DoLinq(List<string> myList)
        {
            var query = from item in myList select item.GetType();
        }
    }
}

```

Reply:
test added